plain_hasher/src/lib.rs

```diff
@@ -30,17 +30,18 @@ impl Hasher for PlainHasher {
 	fn write(&mut self, bytes: &[u8]) {
 		debug_assert!(bytes.len() == 32);
 		let mut bytes_ptr = bytes.as_ptr();
-		let mut prefix_ptr = &mut self.prefix as *mut u64 as *mut u8;
+		let mut prefix_bytes = self.prefix.to_le_bytes();
 
 		unroll! {
```

Comment:
I wonder if the `unroll!` is still needed here – did you try without it? :)

Reply:
I did, it's stll needed :(
```
write_plain_hasher      time:   [1.9603 us 1.9624 us 1.9647 us]                                
                        change: [+381.78% +382.63% +383.51%] (p = 0.00 < 0.05)
                        Performance has regressed.
Found 12 outliers among 100 measurements (12.00%)
  1 (1.00%) low severe
  7 (7.00%) high mild
  4 (4.00%) high severe
```